automation/src/test/scala/org/broadinstitute/dsde/firecloud/api/Orchestration.scala

```diff
@@ -0,0 +1,185 @@
+package org.broadinstitute.dsde.firecloud.api
+
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.firecloud.config.{AuthToken, Config}
+
+import scala.concurrent.duration._
+
+
+trait Orchestration extends FireCloudClient with LazyLogging {
+
+  def responseAsList(response: String): List[Map[String, Object]] = {
+    mapper.readValue(response, classOf[List[Map[String, Object]]])
+  }
+
+  private def retry[T](remainingBackOffIntervals: Seq[FiniteDuration])(op: => () => Boolean): Boolean = {
+    op() match {
+      case true => true
+      case false if remainingBackOffIntervals.isEmpty => false
+      case false if remainingBackOffIntervals.nonEmpty =>
+        Thread sleep remainingBackOffIntervals.head.toMillis
+        retry(remainingBackOffIntervals.tail)(op)
+    }
+  }
+
+  object billing {
+
+    object BillingProjectRole extends Enumeration {
+      val User = Value("user")
+      val Owner = Value("owner")
+    }
+
+    def addUserToBillingProject(projectName: String, email: String, role: BillingProjectRole.Value)(implicit token: AuthToken): Unit = {
+      putRequest(Config.FireCloud.orchApiUrl + s"api/billing/$projectName/${role.toString}/$email")
+    }
+
+    def createBillingProject(projectName: String, billingAccount: String)(implicit token: AuthToken): Unit = {
+      postRequest(Config.FireCloud.orchApiUrl + "api/billing", Map("projectName" -> projectName, "billingAccount" -> billingAccount))
+
+      retry(Seq.fill(30)(10.seconds)) { () =>
```

Comment:
Also if retrying N times with a fixed delay is a common pattern it might be good to define retry as taking a number of attempts (`Int`) and a delay (`FiniteDuration`) which might be simpler to understand.